internal/appsec/dyngo/instrumentation/httpinstr/tags.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package httpinstr
+
+import (
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// setEventSpanTags sets the security event span tags into the service entry span.
+func setEventSpanTags(span ddtrace.Span, events json.RawMessage) {
+	// Set the appsec event span tag
+	// eventTag is the structure to use in the `_dd.appsec.json` span tag.
+	type eventTag struct {
+		Triggers json.RawMessage `json:"triggers"`
+	}
+	// TODO(Julio-Guerra): avoid serializing the json in the request hot path
+	event, err := json.Marshal(eventTag{Triggers: events})
+	if err != nil {
+		log.Error("appsec: unexpected error while serializing the appsec event span tag: %v", err)
+		return
+	}
+	span.SetTag("_dd.appsec.json", string(event))
+	// Keep this span due to the security event
+	span.SetTag(ext.ManualKeep, true)
+	span.SetTag("_dd.origin", "appsec")
+	// Set the appsec.event tag needed by the appsec backend
+	span.SetTag("appsec.event", true)
```

Comment:
Is this tag valuable to the user? If not, why not group it with the rest of `_dd.appsec.event` ? Additionally, doesn't the presence of `_dd.appsec.json` already imply that: yes, it's an event?

Reply:
I had the same question, and the backend team told me it is a retro-compatibility tag we need for now.